### ISSUE TITLE
Obscure update behavior

### ISSUE THREAD

**Issue by fzzr-** [openfl/lime#373]
```shell
$ haxelib upgrade
...
Checking lime
Update lime to 2.2.0 [y/n/a] ? a
Downloading lime-2,2,0.zip...
Download complete : 80085183 bytes in 66.8s (1169.2KB/s)
  Created project/
  Created project/common/
  Install project/common/Extension.cpp
  Install project/common/ExternalInterface.cpp
  Created project/include/
  Install project/include/Utils.h
  Install project/Build.xml
  Install Extension.hx
  Install haxelib.json
  Created dependencies/
  Created dependencies/android/
  Install dependencies/android/project.properties
  Install dependencies/android/build.xml
  Created dependencies/android/src/
  Created dependencies/android/src/org/
  Created dependencies/android/src/org/haxe/
  Created dependencies/android/src/org/haxe/extension/
  Install dependencies/android/src/org/haxe/extension/Extension.java
  Install dependencies/android/AndroidManifest.xml
  Install include.xml
  Current version is now 1.0.0
Done
...
$ haxelib list lime
lime-tools: 1.4.0 1.5.0 1.5.1 1.5.4 1.5.6 [1.5.7]
lime: 0.9.7 0.9.8 1.0.0 1.0.1 2.0.0 2.0.0-alpha.5 2.0.0-alpha.6 2.0.0-alpha.7 2.0.0-alpha.8 2.0.0-beta 2.0.1 2.0.2 2.0.3 2.0.4 2.0.5 2.0.6 2.1.0 2.1.1 2.1.2 [2.1.3]
```

And again and again and again...
```shell
$ haxelib upgrade
...
Checking lime
Update lime to 2.2.0 [y/n/a] ? y
Downloading lime-2,2,0.zip...
```

**Comment by ibilon** [comment]
Do you use haxe 3.2?
https://github.com/HaxeFoundation/haxelib/issues/187

**Comment by fzzr-** [comment]
```shell
$ haxe -version
3.2.0
$ haxelib -version
Unknown command -version
Haxe Library Manager 3.2.0-rc.1 - (c)2006-2015 Haxe Foundation
```

**Comment by MatthijsKamstra** [comment]
I have the same problem on osx.
I removed all installs of lime by hand, and installed again:
```
→ haxelib install lime
Downloading lime-2,2,0.zip...
Download complete : 80085183 bytes in 47.4s (1646.5KB/s)
  Created project/
  Created project/common/
  Install project/common/Extension.cpp
  Install project/common/ExternalInterface.cpp
  Created project/include/
  Install project/include/Utils.h
  Install project/Build.xml
  Install Extension.hx
  Install haxelib.json
  Created dependencies/
  Created dependencies/android/
  Install dependencies/android/project.properties
  Install dependencies/android/build.xml
  Created dependencies/android/src/
  Created dependencies/android/src/org/
  Created dependencies/android/src/org/haxe/
  Created dependencies/android/src/org/haxe/extension/
  Install dependencies/android/src/org/haxe/extension/Extension.java
  Install dependencies/android/AndroidManifest.xml
  Install include.xml
  Current version is now 1.0.0
Done

```


```
→ haxelib list lime
lime-tools: 1.2.9 1.3.2 1.4.0 1.5.4 1.5.6 [1.5.7]
```

now I have a folder named 'unknown' in /usr/lib/haxe/lib/

**Comment by larsiusprime** [comment]
Did you guys run haxelib selfupdate?

**Comment by jgranick** [comment]
Also, if someone wouldn't mind testing:

Download Lime 2.2 from either http://lib.haxe.org or http://openfl.org/builds

Open the zip, and delete the "." files, like ".gitignore" or ".travis.yml"

Try `haxelib local name-of.zip` again. See if this bypasses the problem.

If we have to, we could customize the build to remove those files in order to bypass this haxelib error

**Comment by ibilon** [comment]
@larsiusprime yes but I'm not sure if haxelib got really updated "Haxe Library Manager 3.2.0-rc.1 - (c)2006-2015 Haxe Foundation"

@jgranick I'll try

**Comment by MatthijsKamstra** [comment]
yes I update haxelib... but I have an other problem:
```
→ haxelib
/usr/lib/haxe/std/neko/_std/Sys.hx:87: characters 2-24 : Error chdir /Volumes/Data
tools/haxelib/Main.hx:309: characters 17-43 : Called from
tools/haxelib/Main.hx:1497: characters 2-22 : Called from
--macro:1: character 0 : Called from
```
so perhaps something else is going wrong :(
first install haxelib again...

**Comment by ibilon** [comment]
@jgranick You are right, it was the .files, removing them all work.

**Comment by jgranick** [comment]
I believe there was a commit that added a change so that .files and .directories were ignored, it must be exiting instead.

**Comment by fzzr-** [comment]
@larsiusprime Yes, i did. But currently if i right `haxelib selfupdate` is not working. Anyway it didn't yield results:
```shell
$ haxelib selfupdate
Haxelib was already up to date...
```

**Comment by ibilon** [comment]
@fzzr- was that your first try? When I tried a selfupdate and it failed it still thought it was up to date.

**Comment by fzzr-** [comment]
Yes, first try. (OS X 10.10.3 (14D105g))

**Comment by fzzr-** [comment]
```shell
Checking lime
Downloading lime-2,2,1.zip...
Download complete : 80085332 bytes in 13.4s (5828KB/s)
  Created project/
  Created project/common/
  Install project/common/Extension.cpp
  Install project/common/ExternalInterface.cpp
  Created project/include/
  Install project/include/Utils.h
  Install project/Build.xml
  Install Extension.hx
  Install haxelib.json
  Created dependencies/
  Created dependencies/android/
  Install dependencies/android/project.properties
  Install dependencies/android/build.xml
  Created dependencies/android/src/
  Created dependencies/android/src/org/
  Created dependencies/android/src/org/haxe/
  Created dependencies/android/src/org/haxe/extension/
  Install dependencies/android/src/org/haxe/extension/Extension.java
  Install dependencies/android/AndroidManifest.xml
  Install include.xml
  Current version is now 1.0.0
```

**Comment by ibilon** [comment]
@fzzr- It's a bug with the latest haxelib, there's a fix on github, next release it'll work ;)

**Comment by fzzr-** [comment]
Great! Thx! Now i need to build haxelib :)

**Comment by fzzr-** [comment]
WOW! :+1: It works!
```shell
$ haxelib selfupdate
Haxelib was already up to date...
fzzr:fzzr-.github.io ak$ haxelib version
3.2.0-rc.3
```

**Comment by Type1J** [comment]
I had the same issue, but after a "haxelib selfupdate" it works.

**Comment by jgranick** [comment]
I'll mark this as fixed for now, hopefully the fixed haxelib goes out in the Haxe 3.2 release :wink:

**Comment by wilbefast** [comment]
Hate to necromancer, but I have this issue with haxelib 3.2.0-rc.1

''
[william@GOLIATH ~]$ haxelib version
3.2.0-rc.1
[william@GOLIATH ~]$ haxelib selfupdate
Haxelib was already up to date...
Error writing file /usr/bin/haxelib. Please ensure you have write permissions. 
  [file_open,/usr/bin/haxelib]
[william@GOLIATH ~]$ sudo cp /usr/bin/haxelib /usr/bin/old_haxelib
[william@GOLIATH ~]$ sudo old_haxelib selfupdate
Haxelib was already up to date...
Error compiling haxelib client: Standard library not found
[william@GOLIATH ~]$ sudo su
[root@GOLIATH william]# export HAXE_STD_PATH=/opt/haxe/std
[root@GOLIATH william]# old_haxelib selfupdate
Haxelib was already up to date...
[root@GOLIATH william]# old_haxelib version
[root@GOLIATH william]# old_haxelib install lime
Downloading lime-2,3,0.zip...
Download complete : 87315778 bytes in 71.4s (1193.8KB/s)
  Created project/
  Created project/common/
  Install project/common/Extension.cpp
  Install project/common/ExternalInterface.cpp
  Created project/include/
  Install project/include/Utils.h
  Install project/Build.xml
  Install Extension.hx
  Install haxelib.json
  Created dependencies/
  Created dependencies/android/
  Install dependencies/android/project.properties
  Install dependencies/android/build.xml
  Created dependencies/android/src/
  Created dependencies/android/src/org/
  Created dependencies/android/src/org/haxe/
  Created dependencies/android/src/org/haxe/extension/
  Install dependencies/android/src/org/haxe/extension/Extension.java
  Install dependencies/android/AndroidManifest.xml
  Install include.xml
  Current version is now 1.0.0
Done
''
